pkg/portfwd/client.go

```diff
@@ -130,14 +94,32 @@ func (g GrpcClientRW) Write(p []byte) (n int, err error) {
 func (g GrpcClientRW) Read(p []byte) (n int, err error) {
 	in, err := g.stream.Recv()
 	if err != nil {
-		if errors.Is(err, io.EOF) {
-			return 0, nil
-		}
 		return 0, err
 	}
-	if len(in.Data) == 0 {
-		return 0, nil
-	}
 	copy(p, in.Data)
 	return len(in.Data), nil
 }
+
+func (g GrpcClientRW) Close() error {
+	return g.stream.CloseSend()
+}
+
+func (g GrpcClientRW) LocalAddr() net.Addr {
+	return &net.UnixAddr{Name: "grpc", Net: "unixpacket"}
+}
+
+func (g GrpcClientRW) RemoteAddr() net.Addr {
+	return &net.UnixAddr{Name: "grpc", Net: "unixpacket"}
+}
+
+func (g GrpcClientRW) SetDeadline(t time.Time) error {
+	return nil
```

Comment:
Why we don't implement it? should we log a warning to make the issue visible?

Reply:
Too much logging will become a noise since these are connections. It will be called multiple times and its doesn't help much. SetDeadline is not even usecul in our case as ours is a grpc stream

Reply:
Or just panic. If these functions don't do what they're supposed to (and are never called) a panic would be proof of that.

Reply:
We cannot panic on all cases.

These are left as NO-OP wantedly. These methods may / will be called but its of no value to us as its a grpc tunnel wrapped conn

Reply:
@tamird Please understand the full flow before commenting. All left like that for a reason.

UDP Proxy internally will call SetReadDeadline so we cannot throw error. It will be no-op only. That's why all SetDeadline like methods are returning null.

Reply:
> no-op on SetDeadline is not a reasonable thing to do. It should return an error.

I think you are talking past each other. `SetDeadline()` must exist to implement the interface, but there is not actual implementation of that concept for this particular implementation (there is no deadline). Therefore the method does nothing. It doesn't mean it is an error; just a vestigial method that is not being used.